coffee_core/src/coffee.rs

```diff
@@ -352,10 +352,18 @@ impl PluginManager for CoffeeManager {
             .ok_or_else(|| error!("Repository with name: `{}` not found", repo))?;
 
         let status = repository.upgrade(&self.config.plugins, verbose).await?;
-        for plugins in status.plugins_effected.iter() {
-            self.remove(plugins).await?;
-            self.install(plugins, verbose, false).await?;
+
+        // if status is not up to date, we need to update the plugins as well
+        match status.status {
+            UpgradeStatus::UpToDate(_, _) => {}
+            _ => {
+                for plugins in status.plugins_effected.iter() {
+                    self.remove(plugins).await?;
+                    self.install(plugins, verbose, false).await?;
+                }
+            }
```

Comment:
we can do the oppisite? `UpdateStatus::Upgraded`
```suggestion
            UpdateStatus::Upgraded(_, _) => {
                for plugins in status.plugins_effected.iter() {
                    self.remove(plugins).await?;
                    self.install(plugins, verbose, false).await?;
                }
            },
            _ => {},
```

This is better to avoid bug, if we add a new state we know that only when the plugin is updated we run the remove and install steps

Reply:
I thought about this as well but figured we only wouldn't want to upgrade the plugins if the repository is up to date. 
but I don't have a strong opinion on this one

Reply:
it is better never fail to `_` case, and if we need it is better fall in the case where we do nothing